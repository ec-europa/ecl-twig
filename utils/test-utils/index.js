const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
