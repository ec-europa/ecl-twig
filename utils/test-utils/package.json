{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
