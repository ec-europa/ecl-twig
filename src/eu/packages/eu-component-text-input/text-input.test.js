import path from 'path';
import { renderTwigFile } from '@ecl-twig/test-utils';

// Import data for tests
import dataDefault from '@ecl/eu-specs-text-input/demo/data--default';
import dataDisabled from '@ecl/eu-specs-text-input/demo/data--disabled';
import dataError from '@ecl/eu-specs-text-input/demo/data--with-error';

describe('EU - Text input', () => {
  const template = path.resolve(__dirname, './text-input.html.twig');

  describe('Default', () => {
    test('renders correctly', done => {
      expect.assertions(1);

      renderTwigFile(template, dataDefault, (err, html) => {
        expect(html).toMatchSnapshot();
        done();
      });
    });
  });

  describe('Disabled', () => {
    test('renders correctly', done => {
      expect.assertions(1);

      renderTwigFile(template, dataDisabled, (err, html) => {
        expect(html).toMatchSnapshot();
        done();
      });
    });
  });

  describe('With error', () => {
    test('renders correctly', done => {
      expect.assertions(1);

      renderTwigFile(template, dataError, (err, html) => {
        expect(html).toMatchSnapshot();
        done();
      });
    });
  });
});
