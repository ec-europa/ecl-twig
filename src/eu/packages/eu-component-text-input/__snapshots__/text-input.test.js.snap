// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EU - Text input Default renders correctly 1`] = `
"<div class=\\"ecl-form-group ecl-form-group--text-input\\">
  <label class=\\"ecl-form-label\\" for=\\"example-input-id-1\\"> Label </label>

  <input
    id=\\"example-input-id-1\\"
    name=\\"\\"
    type=\\"text\\"
    placeholder=\\"Placeholder\\"
    class=\\"ecl-text-input\\"
  />
</div>
"
`;

exports[`EU - Text input Disabled renders correctly 1`] = `
"<div class=\\"ecl-form-group ecl-form-group--text-input\\">
  <input
    id=\\"example-input-id-2\\"
    name=\\"\\"
    type=\\"text\\"
    placeholder=\\"Placeholder\\"
    class=\\"ecl-text-input\\"
  />
</div>
"
`;

exports[`EU - Text input With error renders correctly 1`] = `
"<div class=\\"ecl-form-group ecl-form-group--text-input\\">
  <input
    id=\\"example-input-id-3\\"
    name=\\"\\"
    type=\\"text\\"
    placeholder=\\"Placeholder\\"
    class=\\"ecl-text-input\\"
  />
</div>
"
`;
